### PR TITLE
Fix: Scale to fit in 3dp

### PR DESCRIPTION
**Pull request by wen-qing-255**

### CODE REVIEW

File: src/app/models/ModelGroup.js [hunk 1]
```diff
@@ -160,13 +160,21 @@ class ModelGroup extends EventEmitter {
     }
 
     // Calculate selectedGroup's BBox in modelGroup
-    getSelectedModelBBoxDes() {
+    getSelectedModelBBoxDes(docs = true) {
```
Comment (xiekun1992) [inline src/app/models/ModelGroup.js:163]: what does `docs` mean? leave a comment?

**Review by xiekun1992 — APPROVED** [review]

**Comment by jane-rose** [comment]
need rebase master

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -1060,12 +1060,12 @@ export const actions = {
...
         switch (true) {
-            case ['positionX', 'positionY'].some(item => item in transformation):
-                transformMode = 'translate';
-                break;
             case ['scaleX', 'scaleY', 'scaleZ'].some(item => item in transformation):
                 transformMode = 'scale';
                 break;
+            case ['positionX', 'positionY'].some(item => item in transformation):
+                transformMode = 'translate';
+                break;
```
Comment (jane-rose) [inline src/app/flux/printing/index.js:1068]: ？

File: src/app/models/ModelGroup.js [hunk 3]
```diff
@@ -475,29 +483,28 @@ class ModelGroup extends EventEmitter {
...
-            maxObjectPosition.x = Math.max(position.x, maxObjectPosition.x);
-            maxObjectPosition.y = Math.max(position.y, maxObjectPosition.y);
-            maxObjectPosition.z = Math.max(position.z, maxObjectPosition.z);
-
-            minObjectPosition.x = Math.min(position.x, minObjectPosition.x);
-            minObjectPosition.y = Math.min(position.y, minObjectPosition.y);
-        });
+        const boundingBoxTempMax = new Vector3(-Number.MAX_VALUE, -Number.MAX_VALUE, -Number.MAX_VALUE);
+        const boundingBoxTempMin = new Vector3(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE);
+        let boundingBoxTemp = new Box3();
```
Comment (jane-rose) [inline src/app/models/ModelGroup.js:488]: extra `new Box3` `boundingBoxTempMax` `boundingBoxTempMin`

File: src/app/ui/widgets/PrintingMaterial/Material.jsx [hunk 4]
```diff
@@ -63,6 +64,10 @@ function Material({ widgetActions }) {
             dispatch(printingActions.displayModel());
         }
     }
+    function onSelectOfficialDefinition(definitionId) {
```
Comment (jane-rose) [inline src/app/ui/widgets/PrintingMaterial/Material.jsx:67]: unnecessary `onSelectOfficialDefinition` function

File: src/app/models/ModelGroup.js (hunk 1 above)
Comment (jane-rose) [reply src/app/models/ModelGroup.js:163]: Refactor this, add a new funciton return `BoundingBox`

**Pull request closed without merging**